bin/check_security

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+print_usage() {
+    echo "Security Scanner"
+    echo " "
+    echo "Description:"
+    echo "Runs gosec on directories which git detects and marks in the diff."
+    echo "If the branch is detected as 'master', it will scan all"
+    echo "directories regardless of what has been modified locally."
+    echo " "
+    echo "Format:"
+    echo "security_scan [arguments]"
+    echo " "
+    echo "Options:"
+    echo "-h, --help                Show help"
+    echo "-c                        Specify the minimum confidence gosec needs to report an issue."
+    echo "-s                        Specify the minimum severity gosec needs to report an issue"
+    exit 0
+}
+
+# Default values for gosec
+confidence='medium'
+severity='high'
+current_branch=''
+
+while getopts 'b:c:s:h' flag; do
+  case "${flag}" in
+    b) current_branch="${OPTARG}" ;;
+    c) confidence="${OPTARG}" ;;
+    s) severity="${OPTARG}" ;;
+    h) print_usage ;;
+    *) print_usage ;;
+  esac
+done
+
+# This script is designed to be ran within the Secretless-Broker repository,
+# to use elsewhere, you will need to change this.
+working_directory=${PWD}/..
+
+
+# If we are on master, scan the entire repository
+modified_directories="./..."
+
+# Get an array of directories containing modified files
+if [[ ${current_branch} != 'master' ]]
+then 
```

Comment:
nit: not a blocker but we generally used the `if [[ ... ]]; then` pattern instead of the broken-by-lines one